src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -1,17 +1,157 @@
 package com.fasterxml.jackson.core.util;
 
-import com.fasterxml.jackson.core.TokenStreamFactory;
+import java.io.Serializable;
+import java.util.Deque;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Interface for entity that controls creation and possible reuse of {@link BufferRecycler}
  * instances used for recycling of underlying input/output buffers.
  *
  * @since 2.16
  */
-public interface BufferRecyclerPool
-    extends java.io.Serializable
-{
-    public BufferRecycler acquireBufferRecycler(TokenStreamFactory forFactory);
+public interface BufferRecyclerPool extends Serializable {
 
-    public void releaseBufferRecycler(BufferRecycler recycler);
+    BufferRecycler acquireBufferRecycler();
+
+    void releaseBufferRecycler(BufferRecycler recycler);
+
+    /**
+     * Returns the default BufferRecyclerPool implementation which is the thread local based one.
+     */
+    static BufferRecyclerPool defaultRecyclerPool() {
+        return PoolStrategy.THREAD_LOCAL.getPool();
+    }
+
+    enum PoolStrategy {
+        NO_OP(new NonRecyclingRecyclerPool()),
+        THREAD_LOCAL(new ThreadLocalRecyclerPool()), 
+        CONCURRENT_DEQUEUE(new ConcurrentDequePool()),
+        LOCK_FREE(new LockFreePool());
```

Comment:
do we need both CONCURRENT_DEQUEUE and LOCK_FREE? Since users can provide their BufferRecyclerPools, can't we just have a small set of built-in ones?

Reply:
We can keep only one of them if you prefer, but I would keep at least one to have a built-it virtual threads friendly implementation.

Reply:
So, the close could be on the JsonFactory though.

Reply:
I'll have to think about it: I agree about at least one alternative to ThreadLocal (not counting "no-op" one).
I guess due to compact size of implementations I am not super worried about having 2 alternate impls, I guess.